src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -161,7 +162,8 @@ internal async Task<(bool, SumType<CompletionItem[], CompletionList>?)> TryGetPr
 
             await _joinableTaskFactory.SwitchToMainThreadAsync();
 
-            trackingDocumentManager.UpdateVirtualDocument<CSharpVirtualDocument>(documentSnapshot.Uri, new[] { addProvisionalDot }, previousCharacterProjection.HostDocumentVersion);
+            var textChanges = (new[] { addProvisionalDot }).ToVisualStudioTextChangeList();
```

Comment:
It constructs the TextChange just a few lines up. Why not just construct the VS text change type?

Reply:
That would be the right thing to do of course. Changed that.

---
In reply to: [447116142](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447116142) [](ancestors = 447116142)